### ISSUE TITLE
cloud_to_cloud:php interop tests fail on Master

### ISSUE THREAD

**Issue by y-zeng** [grpc/grpc#10576]
https://grpc-testing.appspot.com/job/gRPC_interop_master/12281/

**Comment by stanley-cheung** [comment]
PHP code hasn't changed at all since 1.2.0. Hopefully these timeout are transient issues

**Comment by stanley-cheung** [comment]
I tried to run this locally
```
$ ./tools/run_tests/run_interop_tests.py -l php -s c++ --use_docker -t -j 8
```

And got this error:

```
[HOSTLD]  Linking /var/local/git/grpc/bins/opt/grpc_cpp_plugin
/usr/bin/ld: cannot find -lcares
collect2: error: ld returned 1 exit status
make: *** [/var/local/git/grpc/bins/opt/grpc_cpp_plugin] Error 1
Makefile:14514: recipe for target '/var/local/git/grpc/bins/opt/grpc_cpp_plugin' failed
+ EXITCODE=2
```

Not sure if related.

**Comment by stanley-cheung** [comment]
Ok I narrowed down the regression.

So the PHP tests were passing at this commit: 2dfc3af 4/10 1:09pm PT
And the tests were failing at this commit: 115b4fa 4/10 2:11pm PT

Appears to be related to #10407. That PR moves the `third_party/googletest` and `third_party/protobuf` commit hash and that might be the reason. In particular, for `third_party/protobuf` it goes from the released `3.2.0` tag to an unreleased commit on the `3.3.x` branch.

**Comment by dgquintas** [comment]
Any plan to resolve this?

**Comment by stanley-cheung** [comment]
#10407 is being reverted in #10637. I verified that the PHP interop tests failures are gone in that PR jenkins run.

**Comment by stanley-cheung** [comment]
I am looking at the original issue - ping_pong fails with PHP as client. I narrowed it down to the fact that the PHP client is stuck at the first `read()`  (that's after the first `write()` being successful).

Here's the trace (the last `grpc_completion_queue_pluck` just hangs)
```
I0414 20:59:50.395666980   11261 completion_queue.c:632]     RETURN_EVENT[0x2b9af90]: OP_COMPLETE: tag:0x2c208c8 OK
I0414 20:59:50.395676809   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866ac70)
I0414 20:59:50.395681311   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866ac90)
I0414 20:59:50.395685264   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866acb0)
I0414 20:59:50.395689192   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866acd0)
here 1
I0414 20:59:50.424609823   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0b0)
I0414 20:59:50.424625840   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0d0)
I0414 20:59:50.424629229   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0f0)
I0414 20:59:50.424632075   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b110)
I0414 20:59:50.424652546   11261 call.c:1759]                grpc_call_start_batch(call=0x2c208c8, ops=0x7ffd6866b130, nops=1, tag=0x2c208c8, reserved=(nil))
I0414 20:59:50.424659309   11261 call.c:1402]                ops[0]: SEND_MESSAGE ptr=0x2c21df0
I0414 20:59:50.424670103   11261 completion_queue.c:527]     grpc_completion_queue_pluck(cc=0x2b9af90, tag=0x2c208c8, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0414 20:59:50.425021460   11261 completion_queue.c:224]     grpc_cq_end_op(exec_ctx=0x7ffd6866af30, cc=0x2b9af90, tag=0x2c208c8, error="No Error", done=0x7f1ce2e6a610, done_arg=0x2c39ad8, storage=0x2c39ae0)
I0414 20:59:50.425031240   11261 completion_queue.c:632]     RETURN_EVENT[0x2b9af90]: OP_COMPLETE: tag:0x2c208c8 OK
I0414 20:59:50.425038087   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866b0b0)
I0414 20:59:50.425042451   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866b0d0)
I0414 20:59:50.425046452   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866b0f0)
I0414 20:59:50.425050396   11261 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x7ffd6866b110)
Array
(
    [5] => 1
    [4] => 1
)
I0414 20:59:50.425093302   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0b0)
I0414 20:59:50.425098339   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0d0)
I0414 20:59:50.425102371   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b0f0)
I0414 20:59:50.425106308   11261 metadata_array.c:42]        grpc_metadata_array_init(array=0x7ffd6866b110)
I0414 20:59:50.425111420   11261 call.c:1759]                grpc_call_start_batch(call=0x2c208c8, ops=0x7ffd6866b130, nops=2, tag=0x2c208c8, reserved=(nil))
I0414 20:59:50.425116473   11261 call.c:1402]                ops[0]: RECV_MESSAGE ptr=0x7ffd6866b088
I0414 20:59:50.425120923   11261 call.c:1402]                ops[1]: RECV_INITIAL_METADATA ptr=0x7ffd6866b0f0
I0414 20:59:50.425127853   11261 completion_queue.c:527]     grpc_completion_queue_pluck(cc=0x2b9af90, tag=0x2c208c8, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
```

**Comment by stanley-cheung** [comment]
If I do `GRPC_TRACE=all`, here's the trace

```
I0414 21:05:24.525826207   12184 call.c:1759]                grpc_call_start_batch(call=0x31008c8, ops=0x7ffde6593a80, nops=2, tag=0x31008c8, reserved=(nil))
I0414 21:05:24.525840942   12184 call.c:1402]                ops[0]: RECV_MESSAGE ptr=0x7ffde65939d8
I0414 21:05:24.525852968   12184 call.c:1402]                ops[1]: RECV_INITIAL_METADATA ptr=0x7ffde6593a40
I0414 21:05:24.525870972   12184 client_channel.c:1229]      OP[client-channel:0x3100f48]: [COVERED] RECV_INITIAL_METADATA RECV_MESSAGE
I0414 21:05:24.525886008   12184 http_client_filter.c:427]   OP[http-client:0x3116200]: [COVERED] RECV_INITIAL_METADATA RECV_MESSAGE
I0414 21:05:24.525901118   12184 connected_channel.c:70]     OP[connected:0x3116230]: [COVERED] RECV_INITIAL_METADATA RECV_MESSAGE
D0414 21:05:24.525914190   12184 chttp2_transport.c:1459]    perform_stream_op[s=0x3116710/1]: [COVERED] RECV_INITIAL_METADATA RECV_MESSAGE
D0414 21:05:24.525926465   12184 combiner.c:218]             C:0x3103e10 grpc_combiner_execute c=0x3116d60 cov=1 last=1
D0414 21:05:24.525940172   12184 combiner.c:289]             C:0x3103e10 grpc_combiner_continue_exec_ctx workqueue=0x30ff7f0 is_covered_by_poller=(final=0 elems=1)->1 exec_ctx_ready_to_finish=1 time_to_execute_final_list=0
D0414 21:05:24.525952660   12184 combiner.c:269]             C:0x3103e10 queue_offload --> 0x30ff7f0
I0414 21:05:24.525967028   12184 completion_queue.c:527]     grpc_completion_queue_pluck(cc=0x307af90, tag=0x31008c8, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
D0414 21:05:24.525980847   12184 completion_queue.c:338]     PENDING TAGS: 0x31008c8
D0414 21:05:24.525994059   12184 timer_generic.c:513]        TIMER CHECK SKIP: now_atm=538 min_timer=1006
D0414 21:05:24.526009273   12184 combiner.c:289]             C:0x3103e10 grpc_combiner_continue_exec_ctx workqueue=0x30ff7f0 is_covered_by_poller=(final=0 elems=1)->1 exec_ctx_ready_to_finish=0 time_to_execute_final_list=0
D0414 21:05:24.526022227   12184 combiner.c:308]             C:0x3103e10 maybe_finish_one n=0x3116d60
D0414 21:05:24.526034488   12184 chttp2_transport.c:1224]    perform_stream_op_locked: [COVERED] RECV_INITIAL_METADATA RECV_MESSAGE; on_complete = 0x3116ce0
D0414 21:05:24.526056080   12184 chttp2_transport.c:2467]    FLOW       op: client CREDIT                                  STREAM[1]:incoming_window_delta        0 by                                               add_max_recv_bytes        5 giving 5
D0414 21:05:24.526075346   12184 chttp2_transport.c:2469]    FLOW       op: client CREDIT                                        STREAM[1]:announce_window        0 by                                               add_max_recv_bytes        5 giving 5
D0414 21:05:24.526088260   12184 chttp2_transport.c:1086]    complete_closure_step: 0x3116ce0 refs=0 flags=0x0000 desc=op->on_complete err="No Error"
D0414 21:05:24.526098098   12184 combiner.c:359]             C:0x3103e10 finish old_state=3
D0414 21:05:24.526111218   12184 timer_generic.c:513]        TIMER CHECK SKIP: now_atm=538 min_timer=1006
D0414 21:05:24.526138406   12184 tcp_posix.c:227]            read: error="No Error"
D0414 21:05:24.526159807   12184 tcp_posix.c:232]            READ 0x7f25fc0095c0 (peer=ipv6:[::1]:7074): 17 03 03 00 21 4e f6 05 c4 64 b9 e1 81 85 f1 8b c9 b1 a9 53 36 31 e8 3f dc 7b 3f b3 fc 25 44 2a 90 7b 86 e9 ea 1a 17 03 03 00 25 4e f6 05 c4 64 b9 e1 82 0f 39 83 36 32 b0 d6 17 8f 37 44 b6 dd 7a 19 27 ae 98 6c cf ee a0 c7 ac ad 07 72 e6 66 '....!N...d.........S61.?.{?..%D*.{........%N...d....9.62....7D..z.'..l.......r.f'
D0414 21:05:24.526206631   12184 secure_endpoint.c:145]      READ 0x3108d40: 00 00 12 04 00 00 00 00 00 00 04 00 00 ff ff 00 06 00 00 40 00 fe 03 00 00 00 01 00 00 04 08 00 00 00 00 00 7f ff 00 00 00 00 00 04 01 00 00 00 00 00 00 04 08 00 00 00 00 01 00 00 00 05 '...................@..........................................'
D0414 21:05:24.526223042   12184 combiner.c:218]             C:0x3103e10 grpc_combiner_execute c=0x3111828 cov=0 last=1
D0414 21:05:24.526236623   12184 combiner.c:289]             C:0x3103e10 grpc_combiner_continue_exec_ctx workqueue=0x30ff7f0 is_covered_by_poller=(final=0 elems=0)->0 exec_ctx_ready_to_finish=0 time_to_execute_final_list=0
D0414 21:05:24.526249067   12184 combiner.c:308]             C:0x3103e10 maybe_finish_one n=0x3111828
D0414 21:05:24.526265779   12184 frame_settings.c:228]       CHTTP2:CLI:ipv6:[::1]:7074: got setting INITIAL_WINDOW_SIZE = 65535
D0414 21:05:24.526278068   12184 frame_settings.c:228]       CHTTP2:CLI:ipv6:[::1]:7074: got setting MAX_HEADER_LIST_SIZE = 16384
D0414 21:05:24.526289573   12184 frame_settings.c:228]       CHTTP2:CLI:ipv6:[::1]:7074: got setting GRPC_ALLOW_TRUE_BINARY_METADATA = 1
D0414 21:05:24.526310257   12184 frame_window_update.c:124]  FLOW    parse: client CREDIT                                        TRANSPORT:outgoing_window    38340 by                                                  received_update 2147418112 giving 2147456452
D0414 21:05:24.526339450   12184 frame_window_update.c:114]  FLOW    parse: client CREDIT                                  STREAM[1]:outgoing_window_delta   -27195 by                                                  received_update        5 giving -27190
D0414 21:05:24.526366315   12184 bdp_estimator.c:71]         bdp[ipv6:[::1]:7074]:sched acc=62 est=65536
D0414 21:05:24.526381554   12184 chttp2_transport.c:800]     W:0x31116f0 CLIENT state IDLE -> WRITING [send_ping]
D0414 21:05:24.526393414   12184 combiner.c:407]             C:0x3103e10 grpc_combiner_execute_finally c=0x3111798; ac=0x3103e10; cov=0
D0414 21:05:24.526408657   12184 chttp2_transport.c:2094]    ipv6:[::1]:7074: update initial window size to 88919
D0414 21:05:24.526421052   12184 chttp2_transport.c:800]     W:0x31116f0 CLIENT state WRITING -> WRITING+MORE [push_setting]
D0414 21:05:24.526432889   12184 combiner.c:359]             C:0x3103e10 finish old_state=5
D0414 21:05:24.526451494   12184 combiner.c:289]             C:0x3103e10 grpc_combiner_continue_exec_ctx workqueue=0x30ff7f0 is_covered_by_poller=(final=0 elems=0)->0 exec_ctx_ready_to_finish=0 time_to_execute_final_list=1
D0414 21:05:24.526464139   12184 combiner.c:340]             C:0x3103e10 execute_final[0] c=0x3111798
D0414 21:05:24.526477786   12184 writing.c:218]              W:0x31116f0 CLIENT[1] im-(sent,send)=(1,0) announce=5
D0414 21:05:24.526496222   12184 writing.c:259]              FLOW    write: client DEBIT                                         STREAM[1]:announce_window        5 by                                                         announce        5 giving 0
D0414 21:05:24.526509614   12184 chttp2_transport.c:800]     W:0x31116f0 CLIENT state WRITING+MORE -> WRITING [begin writing]
D0414 21:05:24.526520801   12184 combiner.c:359]             C:0x3103e10 finish old_state=3
D0414 21:05:24.526534084   12184 secure_endpoint.c:276]      WRITE 0x3108d40: 00 00 06 04 00 00 00 00 00 00 04 00 01 5b 57 '.............[W'
D0414 21:05:24.526546168   12184 secure_endpoint.c:276]      WRITE 0x3108d40: 00 00 00 04 01 00 00 00 00 00 00 04 08 00 00 '...............'
D0414 21:05:24.526557607   12184 secure_endpoint.c:276]      WRITE 0x3108d40: 00 00 01 00 00 00 05 '.......'
D0414 21:05:24.526591153   12184 tcp_posix.c:499]            WRITE 0x7f25fc0095c0 (peer=ipv6:[::1]:7074): 17 03 03 00 3d 00 00 00 00 00 00 00 05 65 61 5f 53 76 b0 c8 f8 e2 67 c1 e6 d0 89 63 f1 5a 2c 1c fe 0a e7 66 89 41 22 c5 72 5a 62 b7 1e 5a 59 92 85 9d 94 68 83 b9 81 15 08 e7 86 a7 e5 07 94 a5 a6 d4 '....=........ea_Sv....g....c.Z,....f.A".rZb..ZY....h..............'
D0414 21:05:24.526649877   12184 tcp_posix.c:531]            write: "No Error"
D0414 21:05:24.526665632   12184 combiner.c:218]             C:0x3103e10 grpc_combiner_execute c=0x31117f8 cov=0 last=1
D0414 21:05:24.526679365   12184 combiner.c:289]             C:0x3103e10 grpc_combiner_continue_exec_ctx workqueue=0x30ff7f0 is_covered_by_poller=(final=0 elems=0)->0 exec_ctx_ready_to_finish=0 time_to_execute_final_list=0
D0414 21:05:24.526691632   12184 combiner.c:308]             C:0x3103e10 maybe_finish_one n=0x31117f8
D0414 21:05:24.526703203   12184 chttp2_transport.c:800]     W:0x31116f0 CLIENT state WRITING -> IDLE [finish writing]
D0414 21:05:24.526715355   12184 combiner.c:359]             C:0x3103e10 finish old_state=3
D0414 21:05:24.526727842   12184 timer_generic.c:513]        TIMER CHECK SKIP: now_atm=539 min_timer=1006
D0414 21:05:24.994265407   12184 timer_generic.c:534]        TIMER CHECK BEGIN: now=1493727.100486426 [1006] next=9223372036854775807.000000000 [9223372036854775807] tls_min=1006 glob_min=1006
D0414 21:05:24.994312320   12184 timer_generic.c:440]          .. shard[31]->min_deadline = 1006
```

Then it's just between `TIMER CHECK BEGIN` and `TIMER CHECK END`

**Comment by stanley-cheung** [comment]
Again, if I just re-compile protobuf at 3.2.0 the ping_pong error does not reproduce.

**Comment by dgquintas** [comment]
@sreecha may this also be related to the things you've been looking at lately?

**Comment by stanley-cheung** [comment]
The PHP interop tests errors are gone after #10688 was merged, because the `third_party/protobuf` commit was reverted back to 3.2.0. Removing the Release Blocker tag.

When we do roll forward to protobuf 3.3.0, we will have to pay very close eye again on this.

**Comment by dgquintas** [comment]
Thanks @stanley-cheung !

**Comment by murgatroid99** [comment]
This failure seems to still be present in 1.3.x. See https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/13464/ and https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/13477/

**Comment by stanley-cheung** [comment]
Probably timing. The same fix (#10735) only got merged into `v1.3.x` 13 hours ago (4/19 9:11pm). The two PR jenkins run above were before that time. A more recent one like this: https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/13488/#showFailuresLink seems fine
